mllib/src/test/scala/org/apache/spark/ml/clustering/PowerIterationClusteringSuite.scala

```diff
@@ -76,23 +78,25 @@ class PowerIterationClusteringSuite extends SparkFunSuite
       .setMaxIter(40)
       .setWeightCol("weight")
       .assignClusters(data)
-    val localAssignments = assignments
-      .select('id, 'cluster)
-      .as[(Long, Int)].collect().toSet
-    val expectedResult = (0 until n1).map(x => (x, 1)).toSet ++
-      (n1 until n).map(x => (x, 0)).toSet
-    assert(localAssignments === expectedResult)
+
+    val predictions = Array.fill(2)(mutable.Set.empty[Long])
+    assignments.select("id", "cluster").collect().foreach {
```

Comment:
Nit: I think this was clearer with `.as[(Long,Int)]` as it avoids matching `Row`. I don't feel strongly about it; just less change.

Reply:
Thanks for the suggestion. I have modified.